crates/fuel-core/src/executor.rs

```diff
@@ -963,6 +964,63 @@ mod tests {
         }
     }
 
+    #[test]
+    fn executor_invalidates_expired_tx() {
+        let producer = create_executor(Default::default(), Default::default());
+        let validator = create_executor(Default::default(), Default::default());
+
+        // Given
+        let mut block = test_block(2u32.into(), 0u64.into(), 0);
+
+        let amount = 1;
+        let asset = AssetId::BASE;
+        let mut tx = TxBuilder::new(2322u64)
+            .script_gas_limit(10)
+            .coin_input(asset, (amount as Word) * 100)
+            .coin_output(asset, (amount as Word) * 50)
+            .change_output(asset)
+            .build()
+            .transaction()
+            .clone();
+
+        // When
+        tx.set_expiration(1u32.into());
+        block.transactions_mut().push(tx.clone().into());
+
+        let ExecutionResult {
+            skipped_transactions,
+            mut block,
+            ..
+        } = producer
+            .produce_without_commit(block.into())
+            .unwrap()
+            .into_result();
+
+        // Then
+        assert_eq!(skipped_transactions.len(), 1);
+        assert_eq!(
+            skipped_transactions[0].1,
+            ExecutorError::InvalidTransaction(CheckError::Validity(
+                ValidityError::TransactionExpiration
+            ))
+        );
+
+        // Produced block is valid
+        let _ = validator.validate(&block).unwrap().into_result();
+
+        // Make the block invalid by adding expired transaction
+        let len = block.transactions().len();
+        block.transactions_mut().insert(len - 1, tx.into());
+
+        let verify_error = validator.validate(&block).unwrap_err();
+        assert_eq!(
+            verify_error,
+            ExecutorError::InvalidTransaction(CheckError::Validity(
+                ValidityError::TransactionExpiration
+            ))
+        );
```

Comment:
Should this be two tests?

Reply:
All the execution tests works like that. I followed the current pattern, we can open an issue to change this pattern but it's unrelated to this PR.